### ISSUE TITLE
Rebasing error with the latest tags - update can be closed as 2023-04-18

### ISSUE THREAD

**Issue by dotgitlabhub** [ublue-os/main#146]
When I try to rebase an image with the latest tag like:

rpm-ostree rebase ostree-unverified-registry:ghcr.io/ublue-os/bazzite-desktop:latest

I get this error message:

error: Updating rpm-md repo 'rpmfusion-free-updates-testing': Failed to download gpg key for repo 'rpmfusion-free-updates-testing': Curl error (37): Couldn't read a file:// file for file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-free-fedora-37 [Couldn't open file /etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-free-fedora-37]

Tested rebasing with other images with the latest tags; the same error pops up.

**Comment by castrojo** [comment]
Please paste the output of `rpm-ostree status -v`, thanks!

**Comment by castrojo** [comment]
From someone who had this issue in chat, this seems to be because bazzite is F37 and the rpmfusion keys from the 37 rpmfusion repo got removed. This is why we [recommend not having any layered packages](https://ublue.it/images/) before rebasing.

**Comment by bsherman** [comment]
d the rpmfusion keys from the 37 rpmfusion repo got removed. This is why we [recommend not having any layered packages](https://ublue.it/images/) before rebasing.

Ah ha! It seems funny that this only clicked for me now. I knew having layered packages was problematic, but this is a very specific concrete reason why that is so.

**Comment by dotgitlabhub** [comment]
> Please paste the output of `rpm-ostree status -v`, thanks!

This error occurs on a fresh install with no package layering;

State: idle
AutomaticUpdates: stage; rpm-ostreed-automatic.timer: no runs since boot
Deployments:
● ostree-unverified-image:docker://ghcr.io/ublue-os/silverblue-main:38 (index: 0)
                   Digest: sha256:ad0075647b726da16603020ecf0cfec9d24014882ba8b21517ed422ed322e364
                  Version: 38.20230413.0 (2023-04-14T04:49:18Z)
               BaseCommit: 91c4b4fe1e9e52022c52fd0d06feea0ca9d64d9852aa89d216b58e9cdc962c2a
                   Commit: 8301be490f72be7fafb978072f8072f0a82bc635f8e5374913ec8977015ec297
                           ├─ fedora (2023-04-13T09:32:36Z)
                           ├─ fedora-cisco-openh264 (2023-03-14T10:56:46Z)
                           ├─ fedora-modular (2023-04-13T09:25:16Z)
                           ├─ rpmfusion-free (2023-04-11T14:34:51Z)
                           ├─ rpmfusion-free-updates-testing (2023-04-11T14:24:55Z)
                           ├─ rpmfusion-nonfree (2023-04-11T14:58:51Z)
                           ├─ rpmfusion-nonfree-updates-testing (2023-04-11T14:52:10Z)
                           ├─ updates (2018-02-20T19:18:14Z)
                           ├─ updates-archive (2022-11-11T05:17:33Z)
                           └─ updates-modular (2018-02-20T19:18:14Z)
                   Staged: no
                StateRoot: default
          LayeredPackages: langpacks-en

  ostree-unverified-image:docker://ghcr.io/ublue-os/silverblue-main:38 (index: 1)
                   Digest: sha256:ad0075647b726da16603020ecf0cfec9d24014882ba8b21517ed422ed322e364
                  Version: 38.20230413.0 (2023-04-14T04:49:18Z)
                   Commit: 91c4b4fe1e9e52022c52fd0d06feea0ca9d64d9852aa89d216b58e9cdc962c2a
                StateRoot: default

**Comment by dotgitlabhub** [comment]
> Is this an issue for all images (main, nvidia, and bazzite), or just the bazzite image? If the latter, then please file an issue here:
> 
> https://github.com/ublue-os/bazzite/issues

It's an issue for all images with the latest tags not just bazzite. Read the original bottom comment.

**Comment by castrojo** [comment]
I can't reproduce this but will keep investigating.

**Comment by dotgitlabhub** [comment]
> I can't reproduce this but will keep investigating.

I can't reproduce this error anymore too now. Looks like the issue is fixed with new refreshed container images.

I have been able to successfully rebase to several images with the latest tags.

For reference, my testing was based on a fresh install of kiniote-main:38 off the latest universalblue-38-x86_64-20230416 iso v1.2.1.

rebase journey test were:

rpm-ostree rebase ostree-unverified-registry:ghcr.io/ublue-os/bazzite-desktop:latest

rpm-ostree rebase ostree-unverified-registry:ghcr.io/ublue-os/bluefin:latest

rpm-ostree rebase ostree-unverified-registry:ghcr.io/ublue-os/kinoite-main:latest

rpm-ostree rebase ostree-unverified-registry:ghcr.io/ublue-os/ucore:latest

The only error that was encountered was a bad connection: 

error: Importing: Unencapsulating base: Failed to invoke skopeo proxy method GetBlob: remote error: Get "https://ghcr.io/v2/ublue-os/bazzite-desktop/blobs/sha256:55f0a5c156a35fab701e0d59cd8f87d30e20b9c4cdb6f0322876f18ebb502aad": dial tcp 20.248.137.52:443: i/o timeout

People can ignore this message and just run the rebase command again.

Other issues I found during the rebase test:

just files only worked out of box in bluefin
and just test-cuda failed

output was:
just test-cuda
podman run --user 1000:1000 --security-opt=no-new-privileges --cap-drop=ALL --security-opt label=type:nvidia_container_t  docker.io/mirrorgooglecontainers/cuda-vector-add:v0.1
Trying to pull docker.io/mirrorgooglecontainers/cuda-vector-add:v0.1...
Getting image source signatures
Copying blob 1bc138ea32ad done 
Copying blob 6f94649104a2 done  
Copying blob 6c16e819a84a done  
Copying blob 84b2e9f421b6 done  
Copying blob 9822cda4c699 done  
Copying blob 5d9a20cbabf3 done  
Copying blob ade909bfe2a5 done  
Copying blob e70e5ba470d6 done  
Copying blob ab71e6b7eb90 done  
Copying blob 2f93605342b5 done  
Copying blob 925740434ebd done  
Copying blob fe61ad4992f7 done  
Copying config 57146f1c77 done  
Writing manifest to image destination
Storing signatures
Error: OCI runtime error: crun: `/proc/self/attr/keycreate`: OCI runtime error: unable to process security attribute
error: Recipe `test-cuda` failed on line 19 with exit code 126

in ucore:

tailscale and cockpit was installed but services were not enabled by default

some cockpit modules were missing such as:
cockpit-kdump cockpit-machines cockpit-ostree cockpit-pcp cockpit-sosreport
